### PR TITLE
[FIX] OWMosaic: Fix crash for attribute with no values

### LINKED ISSUE #1940: ValueError:  minlength must be positive
https://sentry.io/biolab/orange3/issues/206599981/

```
ValueError:  minlength must be positive
(5 additional frame(s) were not displayed)
...
  Module "Orange.statistics.distribution", line 295, in get_distribution

    return Discrete(dat, variable, unknowns)
  Module "Orange.statistics.distribution", line 39, in __new__

    return cls.from_data(dat, variable)
  Module "Orange.statistics.distribution", line 63, in from_data

    dist, unknowns = data._compute_distributions([variable])[0]
  Module "Orange.data.table", line 1314, in _compute_distributions

    dist, unknowns = bincount(m, len(var.values) - 1, W)
  Module "Orange.statistics.util", line 52, in bincount

    minlength=minlength).astype(float),
```

### PR DESCRIPTION
**Pull request by VesnaT**
##### Issue

Fixes #1940

##### Includes
- [X] Code changes
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by astaric** [comment]
Not sure how to test this. What I tried was adding a discrete variable with all missing values (using feature constructor) and using that in mosaic. It still crashed, but it might be a different problem.

**Comment by VesnaT** [comment]
Looks like that is another problem.

To test this use an existing dataset with feature with all missing values, since the problem occurred when the variable's "values" is empty.
It is impossible to construct discrete feature with no values with 
feature constructor.

File: Orange/widgets/visualize/owmosaic.py [hunk 1]
```diff
@@ -899,6 +899,13 @@ def draw_legend(x0_x1, y0_y1):
             self.Warning.no_valid_data.clear()
 
         if self.interior_coloring == self.PEARSON:
+            attrs = [attr for attr in attr_list if not data.domain[attr].values]
+            if attrs:
+                CanvasText(self.canvas,
+                           "Feature {} has no values".format(attrs[0]),
+                           (self.canvas_view.width() - 120) / 2,
+                           self.canvas_view.height() / 2)
+                return
```
Comment (janezd) [inline Orange/widgets/visualize/owmosaic.py]: I would put this before `if`, that is, fail in both cases, with Pearson or with class distribution. The visualization does not make sense if any attribute has no values.

Comment (janezd) [inline Orange/widgets/visualize/owmosaic.py]: screen shot 2017-01-24 at 21 13 31

File: Orange/widgets/visualize/owmosaic.py [hunk 2]
```diff
@@ -948,6 +955,12 @@ def get_max_label_width(attr):
                 get_conditional_distribution(subset, attr_list)
 
         # draw rectangles
+        if self.interior_coloring != self.PEARSON and not class_var.values:
+            CanvasText(self.canvas,
+                       "Class variable {} has no values".format(class_var),
+                       (self.canvas_view.width() - 140) / 2,
+                       self.canvas_view.height() / 2)
+            return
```
Comment (janezd) [inline Orange/widgets/visualize/owmosaic.py]: And then you can remove this.

**Comment by janezd** [comment]
@astaric, try this data: [x.tab.zip](https://github.com/biolab/orange3/files/727860/x.tab.zip)

**Comment by lanzagar** [comment]
I am merging this, but two bugs in Mosaic still remain (unrelated to this PR):
- using `anneal` data and choosing feature `bt` crashes (I think this is what @astaric noticed)
- Find Informative Mosaics (also on anneal) does not sort correctly (nan scores incomparable) - I think @janezd fixed the same issue in another widget's vizrank?

**Pull request merged**